### 231210 까지의 수정사항

jihyun11 opened the pull request:
+ update 기능 추가하였습니다.
+ (update는 delete와 다르게 '사용자가 수정한 내용'을 매개변수로 받으므로 type에서 hidden을 쓰면 안됨)
+ 위의 기능을 update 페이지를 만들어서 따로 뺄 예정입니다.

jihyun11 committed:
### 231210 까지의 수정사항
+ update 기능 추가하였습니다.
+ (update는 delete와 다르게 '사용자가 수정한 내용'을 매개변수로 받으므로 type에서 hidden을 쓰면 안됨)
+ 위의 기능을 update 페이지를 만들어서 따로 뺄 예정입니다.

jabel123 commented on src/main/java/com/jihyun/myboard/service/PostService.java:
```diff
@@ -30,7 +30,10 @@ public void deletePost(String content, String writer) {
-    // 이곳에 게시글 수정 메소드를 작성합니다.
+    public void updatePost(String content, String writer, String id) {
+        postMapper.updatePost(content, writer, id);
+        System.out.println("업데이트 서비스");
```
클래스위에 @Slfj4 라는 어노테이션을 쓰면

log.info("업데이트 서비스");
이런식으로 로그를 남길수 있는데, System.out.println 보단, 보통 이런식으로 로그를 남깁니다. 
이유는 log4j에 대해 공부해보시면 알 수 있으실 거에요 😄

jihyun11 committed:
### 231213 까지의 수정사항
+ 번역기 기능 추가하였습니다. (언어 선택 가능)
+ 가타카나 퀴즈 맞히기 기능 추가하였습니다. (랜덤으로 보기 선별 및 점수 카운트 가능)

jabel123 commented on src/main/java/com/jihyun/myboard/service/ApiService.java:
```diff
@@ -0,0 +1,101 @@
+package com.jihyun.myboard.service;
+
+//import org.h2.util.json.JSONObject;
```
안쓰는 코드는 제거

jabel123 commented on src/main/java/com/jihyun/myboard/service/ApiService.java:
```diff
@@ -0,0 +1,101 @@
+    private static final String CLIENT_SECRET = "XlgDyxer6x"; // 클라이언트 시크릿
+    private static final String API_URL = "https://openapi.naver.com/v1/papago/n2mt";
+
+    String a;
```
해당 변수를 속성으로 선언하지 않고, 사용처인 23번 라인 근처에 변수를 선언하는게 좋을것 같습니다 😄

jabel123 commented on src/main/java/com/jihyun/myboard/service/ApiService.java:
```diff
@@ -0,0 +1,101 @@
+
+        System.out.println(translatedText);
+        return translatedText;
+
```
불필요한 띄어쓰기가 없어도 좋을것 같습니다

jabel123 commented on src/main/java/com/jihyun/myboard/service/ApiService.java:
```diff
@@ -0,0 +1,101 @@
+
+    public static String post(String apiUrl, Map<String, String> requestHeaders, String text, String srcLang, String targetLang) {
+        HttpURLConnection con = connect(apiUrl);
+        String postParams = "source="+srcLang+"&target="+targetLang+"&text=" + text; //원본언어: 한국어 (ko) -> 목적언어: 영어 (en)
```
'+' 사이사이에 일관성있게 코드가 띄워지면 좋을것 같습니다

jabel123 commented on src/main/java/com/jihyun/myboard/service/ApiService.java:
```diff
@@ -0,0 +1,101 @@
+            URL url = new URL(apiUrl);
+            return (HttpURLConnection) url.openConnection();
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("API URL이 잘못되었습니다. : " + apiUrl, e);
```
아래와 같은 느낌으로, Custom한 Exception클래스를 사용하시면 좋을것 같습니다.
그리고, log.error를 이용하여 로그도 남겨주시면 좋을것 같습니다.
            log.error(e.getMessage(), e);
            throw new MyException("인코딩 실패");

jabel123 reviewed (approved):
수고하셨습니다!
